include/zephyr/net/socket.h

```diff
@@ -890,38 +921,6 @@ static inline ssize_t recv(int sock, void *buf, size_t max_len, int flags)
 	return zsock_recv(sock, buf, max_len, flags);
 }
 
-/** @cond INTERNAL_HIDDEN */
-/*
- * Need this wrapper because newer GCC versions got too smart and "typecheck"
- * even macros, so '#define fcntl zsock_fcntl' leads to error.
- */
-static inline int zsock_fcntl_wrapper(int sock, int cmd, ...)
-{
-	va_list args;
-	int flags;
-
-	va_start(args, cmd);
-	flags = va_arg(args, int);
-	va_end(args);
-	return zsock_fcntl(sock, cmd, flags);
-}
-
-#define fcntl zsock_fcntl_wrapper
-/** @endcond */
-
-/** POSIX wrapper for @ref zsock_ioctl */
-static inline int ioctl(int sock, unsigned long request, ...)
-{
-	int ret;
-	va_list args;
-
-	va_start(args, request);
-	ret = zsock_ioctl(sock, request, args);
-	va_end(args);
-
-	return ret;
-}
-
 /** POSIX wrapper for @ref zsock_sendto */
 static inline ssize_t sendto(int sock, const void *buf, size_t len, int flags,
```

Comment:
I wonder, could we remove those duplicate wrappers already, and just use the POSIX headers below in case `CONFIG_NET_SOCKETS_POSIX_NAMES` is enabled as well?

Reply:
If one sets `CONFIG_NET_SOCKETS_POSIX_NAMES=y` and `CONFIG_POSIX_API=n`, then there is a compilation error if these symbols are not in `socket.h`. It looks like we are stuck with these until the `NET_SOCKETS_POSIX_NAMES` is removed from the Kconfig file, and then we can remove the settings.